go.mod

```diff
@@ -52,6 +51,7 @@ require (
 // This looks "horrible", but is due to the Rook including k8s.io/kubernetes directly which is not recommended,
 // ee https://github.com/kubernetes/kubernetes/issues/79384#issuecomment-505725449
 replace (
+	github.com/hashicorp/vault/sdk => github.com/hashicorp/vault/sdk v0.1.14-0.20201110190005-0340b40575ad
```

Comment:
Why do we need to pin it to this specific version? I thought if we updated the modules, it should resolve the latest automatically.

Reply:
IIUC, vault/sdk is not a directly inherited module in rook. It is coming in as a dependency of github.com/libopenstorage/secrets. 

As per https://research.swtch.com/vgo-principles the recommended way to fix this error is a two step process:
1. Bump up the version of 'github.com/hashicorp/vault' in 'github.com/libopenstorage/secrets' such that it does not reference the deleted commit anymore.
2. Bump up the version of 'github.com/libopenstorage/secrets' in Rook to the commit which performed step 1.

In cases where changes like in step 1 aren't possible(If the repo is not well maintained etc), we can use the replace section of go.mod to override the version like in this PR.

When I first looked at this PR, I missed that libopenstorage/secrets is also maintained by Rook developers. I suggest using the prescribed method and not the replace option.

Reply:
I thought libopenstorage/secrets maintained by protwox folks, not by rook community